pkg/object/objects.go

```diff
@@ -301,7 +302,10 @@ func (os K8sObjects) Sort(score func(o *K8sObject) int) {
 func (os K8sObjects) ToMap() map[string]*K8sObject {
 	ret := make(map[string]*K8sObject)
 	for _, oo := range os {
-		ret[oo.Hash()] = oo
+		// ignore comments
+		if oo.Kind != "" || oo.Name != "" || oo.Namespace != "" {
```

Comment:
how about putting this into a helper function like (o *K8sObject) empty() bool{} ?
this will also make the code self-documenting.